src/message.js

```diff
@@ -0,0 +1,403 @@
+//misc imports
+import { PowerLevelAction } from "matrix-bot-sdk";
+import { createRequire } from "node:module";
+const require = createRequire(import.meta.url);
+
+import { Sendjson } from "./sendjson.js";
+
+// commands
+import { Uptime } from "./commands/uptime.js";
+import { Join } from "./commands/join.js";
+import { Leave } from "./commands/leave.js"; //does blacklisting
+import { Unblacklist } from "./commands/unblacklist.js";
+import { Restart } from "./commands/restart.js";
+import { Mute } from "./commands/mute.js";
+import { Banlist } from "./commands/banlist.js";
+import { FollowBanList } from "./commands/followbanlist.js";
+import { Rules } from "./commands/rules.js";
+import { CC } from "./commands/create-community.js";
+
+const sendjson = new Sendjson();
+
+class message {
+  constructor(logRoom, commandRoom, config, authorizedUsers) {
+    //map to relate scams and their responses (for deletion)
+    this.tgScamResponses = new Map();
+    this.tgScamReactions = new Map();
+
+    //config thingys
+    this.logRoom = logRoom;
+    this.commandRoom = commandRoom;
+    this.config = config;
+    this.authorizedUsers = authorizedUsers;
+
+    //fetch keywords
+    this.keywords = require("../keywords.json");
+
+    //create collection of different commands to run
+    this.commands = new Map();
+
+    this.commands.set("uptime", new Uptime());
+    this.commands.set("join", new Join());
+    this.commands.set("leave", new Leave());
+    this.commands.set("unblacklist", new Unblacklist());
+    this.commands.set("restart", new Restart());
+    this.commands.set("mute", new Mute());
+    this.commands.set("banlist", new Banlist());
+    this.commands.set("followbanlist", new FollowBanList());
+    this.commands.set("rules", new Rules());
+    this.commands.set("create-community", new CC());
+  }
+
+  // async run ({client, roomId, event, mxid, displayname, blacklist}){
+  async run(datapoints) {
```

Comment:
i honestly couldn't figure out how to break this down to make an example protection, it seems like there's so many different things being done in this one handle